src/components/v2v3/V2V3Project/V2V3ProjectSettings/pages/NewEditCyclePage/PayoutsSection/PayoutsTable/PayoutSplitRow.tsx

```diff
@@ -74,26 +99,41 @@ export function PayoutSplitRow({
   ]
 
   return (
-    <PayoutsTableRow className="text-primary text-sm">
-      <Cell className="py-6">
-        <PayoutTitle payoutSplit={payoutSplit} />
-      </Cell>
-      <Cell className="py-6">
-        <div className="flex items-center gap-3">
-          <FormattedNumberInput
-            accessory={
-              <span className="text-sm">
-                {V2V3_CURRENCY_METADATA[getV2V3CurrencyOption(currency)].symbol}
-              </span>
-            }
-            accessoryPosition="left"
-            value={formattedAmount}
-            onChange={onChange}
-            className="h-10"
-          />
-          <PopupMenu items={menuItems} />
-        </div>
-      </Cell>
-    </PayoutsTableRow>
+    <>
+      <PayoutsTableRow className="text-primary text-sm">
+        <Cell className="py-6">
+          <PayoutTitle payoutSplit={payoutSplit} />
+        </Cell>
+        <Cell className="py-6">
+          <div className="flex items-center gap-3">
+            <FormattedNumberInput
+              accessory={
+                <span className="text-sm">
+                  {
+                    V2V3_CURRENCY_METADATA[getV2V3CurrencyOption(currency)]
+                      .symbol
+                  }
+                </span>
+              }
+              accessoryPosition="left"
+              value={formattedAmount}
+              onChange={onAmountPercentageInputChange}
+              className="h-10"
+            />
+            <PopupMenu items={menuItems} />
+          </div>
+        </Cell>
+      </PayoutsTableRow>
+      <AddEditAllocationModal
+        allocationName="payout"
+        availableModes={
+          new Set([distributionLimitIsInfinite ? 'percentage' : 'amount'])
+        }
+        editingData={addEditAllocationModalEntity}
```

Comment:
Logging `addEditAllocationModalEntity` has all the data defined and correct, but then inside `AddEditAllocationModal`, some data (particularly `projectId` and `amount`) and undefined for a period of time. Any ideas?

Reply:
how did you go trying the `setTImeout` trick?